lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -193,6 +193,16 @@ Type CompleteGenericTypeResolver::resolveDependentMemberType(
   // base type into it.
   auto concrete = ref->getBoundDecl();
   tc.validateDeclForNameLookup(concrete);
+
+  if (auto typeAlias = dyn_cast<TypeAliasDecl>(concrete)) {
+    if (auto protocol = dyn_cast<ProtocolDecl>(typeAlias->getDeclContext())) {
+      // We need to make sure the generic environment of a surrounding protocol
+      // propagates to the typealias, since the former may not have existed when
+      // the typealiases type was first computed.
+      // FIXME: See the comment in the ProtocolDecl case of validateDecl().
+      tc.validateDecl(protocol);
```

Comment:
What happens if you just unconditionally call validateDecl() and remove the validateDeclForNameLookup() call above?